stubs/SQLAlchemy/sqlalchemy/util/_collections.pyi

```diff
@@ -1,21 +1,27 @@
 import collections.abc
 import sys
-from typing import Any
+from _typeshed import Self
+from collections.abc import Callable, Iterable, Iterator
+from typing import Any, Generic, NoReturn, TypeVar, overload
 
 from ..cimmutabledict import immutabledict as immutabledict
+from ..sql.elements import ColumnElement
+
+_S = TypeVar("_S")
+_T = TypeVar("_T")
 
 collections_abc = collections.abc
 
-EMPTY_SET: Any
+EMPTY_SET: frozenset[Any]
 
 class ImmutableContainer:
-    __delitem__: Any
-    __setitem__: Any
-    __setattr__: Any
+    def __delitem__(self, *arg: object, **kw: object) -> NoReturn: ...
```

Comment:
Would setting the arguments to NoReturn work better with typecheckers?

Reply:
The arguments?

Reply:
Yes, calling a function with NoReturn argument will generally cause type checkers to show an error, which is what we'd want here.

Reply:
Considering the necessary `# type: ignore`s and the hacky nature of this, I'd prefer to keep using `object` (or `Any`).